src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -70,8 +70,8 @@ public class DistributedSession : ISession
             _idleTimeout = idleTimeout;
             _ioTimeout = ioTimeout;
             _tryEstablishSession = tryEstablishSession;
-            _store = new Dictionary<EncodedKey, byte[]>();
             _logger = loggerFactory.CreateLogger<DistributedSession>();
+            _store = new Dictionary<EncodedKey, byte[]>();
```

Comment:
What if you moved this here instead:
https://github.com/hishamco/Session/blob/06b2054f1761803c7de7b8b32370df36cdd32471/src/Microsoft.AspNetCore.Session/DistributedSession.cs#L181

Reply:
Yes, in that else block.

Reply:
This is a good option too